datafusion/physical-expr/src/equivalence/ordering.rs

```diff
@@ -207,19 +212,6 @@ impl IntoIterator for OrderingEquivalenceClass {
     }
 }
 
-/// This function constructs a duplicate-free `LexOrdering` by filtering out
```

Comment:
Despite the function being `pub` it appears to be in a non pub module, so is not exposed. You can see that it is not present in the docs:

https://docs.rs/datafusion/latest/datafusion/index.html?search=collapse_lex_ordering

(when I marked it deprecated then clippy complained it was dead code)